extract.js

```diff
@@ -144,7 +144,14 @@ function loadBabelOpts(opts) {
 
 		if (Array.isArray(fileOpts[key]) && Array.isArray(opts.parserOpts[key])) {
 			// combine arrays for plugins
-			opts.parserOpts[key] = opts.parserOpts[key].concat(fileOpts[key]);
+			// plugins in fileOpts could be string, array or object
+			for (const plugin of fileOpts[key]) {
+				if (Array.isArray(plugin) || typeof plugin === 'string') {
+					opts.parserOpts[key].push(plugin);
+				} else {
+					opts.parserOpts[key].push([plugin.key, plugin.options]);
```

Comment:
[ask] The original code uses `.concat` not to mutate the `opts.parserOpts[key]` array, while the changed code uses `.push` to mutate the array. No problems?

Reply:
Done.